python-frontend/src/main/java/org/sonar/python/types/InferredTypes.java

```diff
@@ -62,4 +66,12 @@ public static InferredType or(InferredType t1, InferredType t2) {
     return InferredTypes.anyType();
   }
 
+  public static InferredType declaredType(TypeAnnotation typeAnnotation) {
+    Expression expression = typeAnnotation.expression();
+    if (expression.is(Kind.NAME) && !((Name) expression).name().equals("Any")) {
+      // TODO change it to DeclaredType instance
```

Comment:
I didn't create `DeclareType` class in this PR because I don't see a concrete benefit for the rule we have at the moment. What do you think?

Reply:
I think that we may have false positives if the Typeshed declarations we import contain a function which can return a subtype of its declared return type.
```
class A:
  pass
class A1(A):
  pass
def f() -> A: ...
```
```
a = A1()
if f() is a: # FP would be raised here
  pass
```

Reply:
As discussed, we will handle this in a separate ticket: https://jira.sonarsource.com/browse/SONARPY-605